### ISSUE TITLE
Duplicate empty endpoints groups

### ISSUE THREAD

**Issue by kszafran** [a85/POSTMan-Chrome-Extension#10277]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the Issue

We have a collection in a team workspace, which we update every now and then (using the PUT endpoint in the Postman API). People who use the Postman Desktop app have a problem, where old endpoint groups sometimes don't disappear, but become empty and stay:
<img width="269" alt="Screen Shot 2021-09-03 at 15 13 12" src="https://user-images.githubusercontent.com/3056812/132010862-d97bbfe4-a07f-48d9-bca4-b1a6a9957245.png">
The list of groups gets pretty long and you have to waste time looking for a group, which actually has some endpoints. In the JSON those empty groups looks like this:
```
		{
			"name": "Users",
			"item": []
		},
```

I though it only affects the desktop app, but not the one at postman.co. However, when I export the collection from the web client, it also includes the empty groups. They are just not shown in the web client.

### Steps To Reproduce

1. Create a large collection (ours has 45 endpoint groups, ~250 endpoints total)
2. Upload it to a team workspace using the Postman API (https://documenter.getpostman.com/view/631643/JsLs/#99810ef3-3cc0-a6cc-06f5-d8e2ae9d84e4)
3. Open it in the desktop app (sync on).
4. Update it a few times using the Postman API (https://documenter.getpostman.com/view/631643/JsLs/#aa3701e8-7f99-b421-7d74-0d571b051f3c)
5. Check the collection in the desktop app. It should have empty endpoint groups now. 

### Environment Information

```markdown
- Operating System: macOS 11.5.1 in my case (but people on other systems also have this problem)
- Platform Type: Postman API (the problem is visible in Native App, but not in Postman Web)
- Postman Version: 8.12.0 (it's not a new problem though, we've had it for at least a year or two)
```

**Comment by postman-staff** [comment]
Hi @kszafran,

We checked this from our end and it looks like the primary cause of this is due to the number of requests involved and sync. [Here is the screen-record for reference](https://www.loom.com/share/da7298cf4b404f13863983d120132277) on how we replicated and as you see from recording inching 2 min, we exactly see the error you've mentioned.

So first we tried creating a new request "GetPoint" on the first folder "00" - the folder including all, does reduce to zero and then populates with the record we added.

Next, we deleted the request "003" from the folder "00" and the same behavior was seen but was updated as expected.

In this, we specified that while the syncing is in progress, the requests zero out for the group and then gradually populate back once the sync is completed. I believe this is the same occurrence in your case as well, do try waiting for the sync action to be completed (few additional seconds if that would help) to see if the entered details reflect right and only then export to avoid any loss of data/endpoints.

I would also suggest trying this on a smaller collection or breaking the collection into smaller bits and see if that helps you get around this situation.

via @Raj-postman | Postman Support

**Comment by kszafran** [comment]
Thank you for looking into this. In our case it's not related to sync, at least not directly. The status is "In Sync" and the empty groups are still there. Also, If I export the collection from Postman Web (where I don't see the duplication), the exported JSON also contains those empty groups (as described in the original post), so it seems to be a problem with Postman API, not just the desktop app.

> I would also suggest trying this on a smaller collection or breaking the collection into smaller bits and see if that helps you get around this situation.

We have already deleted almost all descriptions from our collections, otherwise we were getting random 500s when calling PUT (our older issue: https://github.com/postmanlabs/postman-app-support/issues/8213). Splitting the collection would be delving even more into workarounds and making our documentation less useful for developers. I really hope that Postman can fix this problem and those 500s. I enjoy using Postman, it's just those two significant problems that make us consider switching to another tool.

If you're not able to reproduce this error I may be able to share our collection over some private channel for troubleshooting.

**Comment by postman-staff** [comment]
@kszafran True I understand that. The issue with updating using Postman API is currently being tracked internally as well, the team should post an update in the [GitHub thread](https://github.com/postmanlabs/postman-app-support/issues/8213) once we have any improvements on the same.

For the empty sub-folder generations - Sure, you may share the collection with us by submitting a ticket to us via https://www.postman.com/support.

Two things in addition to that which can help there, is the shared collection the one you have exported after updating using the API?
1) If yes, then please share the original collection which was there prior to the update, this should help us compare the differences.
2) What exactly was updated in the collection? To see if there are edge cases that could be branching this.

It would also be great if you could share a **screen record (**_We find [loom.com](loom.com) helpful for this_**)** similar to one I've shared earlier - this would help us replicate as well as support engineering to identify the cause of the error. Please share these details via [Ticket](https://www.postman.com/support) raised as mentioned above to keep this private :) Looking forward to your response.

via @Raj-postman | Postman Support

**Comment by kszafran** [comment]
I have shared our collections in support ticket #83984. I have also included a full collection that causes the frequent 500s reported in https://github.com/postmanlabs/postman-app-support/issues/8213.

**Comment by postman-staff** [comment]
@kszafran Thank you for sharing the required details. We will process this request next via ticket.

via @Raj-postman | Postman Support

**Comment by kszafran** [comment]
After investigating this problem, it seems that it only happens if we PUT on the same collection multiple times concurrently. While the Postman team is working on this, I have modified our CI pipeline to prevent such concurrent updates.

**Comment by kszafran** [comment]
The problem still occurs even when our CI pipeline never makes concurrent updates, although it is much more rare then. I don't think I would be able to reproduce it reliably.

**Comment by davidespihernandez** [comment]
Hi, @kszafran!
I'm closing this ticket because I believe it's being tracked in another place.

However, in the last months, we have added some new APIs to the Postman API that make easier work with big collections:
- [Collections Items API](https://www.postman.com/postman/workspace/postman-public-workspace/folder/12959542-efeda219-66e1-474c-a83b-253d15723bf7). It allows you to CRUD requests, responses and folders in a collection, individually, without having to do a costly PUT of the whole collection.
- In short (early November 2023) we will expose new APIS to copy or move responses, requests, and folders between collections (or in the same collection). This will also avoid the need to do a full PUT.

Please check the [Postman API public collection](https://www.postman.com/postman/workspace/postman-public-workspace/collection/12959542-c8142d51-e97c-46b6-bd77-52bb66712c9a) documentation, where we are constantly adding new improvements.

**Comment by kszafran** [comment]
Collections API sounds interesting, though it might not quite fit our use case. We rebuild/regenerate the Postman collection when our source API documentation changes and just upload a new file. To take advantage of Collections API we'd need to diff the new collection against the old one and make incremental updates. That's possible, but not trivial and error prone.
